src/Buffer.mo

```diff
@@ -890,6 +994,19 @@ module {
   /// The deterministic hash function is a function of the elements in the Buffer, as well
   /// as their ordering.
   ///
+  /// Example:
+  /// ```motoko include=initialize
+  /// import Nat "mo:base/Nat";
+  /// import Nat32 "mo:base/Nat32";
+  ///
+  /// buffer.add(1);
+  /// buffer.add(2);
+  /// buffer.add(3);
+  /// buffer.add(1000);
+  ///
+  /// Buffer.hash<Nat>(buffer, Nat32.fromNat);
```

Comment:
```suggestion
  /// import Hash "mo:base/Hash";
  ///
  /// buffer.add(1);
  /// buffer.add(2);
  /// buffer.add(3);
  /// buffer.add(1000);
  ///
  /// Buffer.hash<Nat>(buffer, Hash.hash);
```

Sub-optimally, the hash function for `Nat` is not in the `Nat` module but is in `Hash`. We should probably move this